docs/bookstore/page-1/pg1-review-svc-1.md

```diff
@@ -0,0 +1,379 @@
+
+```javascript
+app.post('/add', async (req, res) => {
+  try {
+    const receivedEvent = HTTP.toEvent({headers: req.headers, body: req.body});
+    const brokerURI = process.env.K_SINK;
+
+    if (receivedEvent.type === 'new-review-comment') {
+      const response = await fetch(brokerURI, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/cloudevents+json',
+          'ce-specversion': receivedEvent.specversion,
+          'ce-type': receivedEvent.type,
+          'ce-source': receivedEvent.source,
+          'ce-id': receivedEvent.id,
+        },
+        body: JSON.stringify(receivedEvent.data),
+      });
+    }
+  } catch (err) {
+    console.error(err);
+  }
+});
+```
+
+**Receiving Events**: The `/add` endpoint receives events from the frontend. It uses CloudEvents' SDK to convert the incoming request into a CloudEvent object:
+
+```javascript
+const receivedEvent = HTTP.toEvent({headers: req.headers, body: req.body});
+```
+
+**Determining Broker Address**: The broker's address is dynamically assigned in the cluster. The Node.js server retrieves this address from the environment variable `K_SINK`:
+
+```javascript
+const brokerURI = process.env.K_SINK;
+```
+
+You may wonder who told the environment variable about the address? That’s Knative SinkBinding.
+
+**Event Filtering**: The service checks the event type. If it's a `new-review-comment`, it forwards the event to the broker:
+
+```javascript
+if (receivedEvent.type === 'new-review-comment') {
+  // logic that forwards the event, see below
+}
+```
+
+**Forwarding Events Logic**: The event is forwarded to the broker with the appropriate CloudEvent headers:
+
+```javascript
+const response = await fetch(brokerURI, {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/cloudevents+json',
+    'ce-specversion': receivedEvent.specversion,
+    'ce-type': receivedEvent.type,
+    'ce-source': receivedEvent.source,
+    'ce-id': receivedEvent.id,
+  },
+  body: JSON.stringify(receivedEvent.data),
+});
+```
+
+![Image](images/image21.png)
+
+**Exploring Other Functions**
+
+The Node.js server contains other functions that follow a similar pattern, with detailed comments explaining their functionality:
+
+- `/insert`: Receives CloudEvents and inserts the payload into the PostgreSQL database.
+- `/comment`: Creates a WebSocket connection with the frontend to transport comments from the database.
+
+### **Step 2: Create Broker**
+
+![Image](images/image17.png)
+
+The Broker acts as a router in your event-driven application, receiving events and routing them to the correct destination.
+
+
+
+- 1: Create a new YAML file named `node-server/config/200-broker.yaml` and add the following content:
+
+???+ abstract "_node-server/config/200-broker.yaml_"
+    ```yaml
+    apiVersion: eventing.knative.dev/v1
+    kind: Broker
+    metadata:
+    name: bookstore-broker
+    ```
```

Comment:
and regarding metadata.name:

--> 
```
apiVersion: eventing.knative.dev/v1
kind: Broker
metadata:
  name: bookstore-broker
```

Reply:
Ohh I see! In [this part of the code, which is non-collapsible](https://deploy-preview-6007--knative.netlify.app/docs/bookstore/page-4/pg4-sequence/#step-0-learn-sequence), used for teaching purposes, readers don't need to copy and paste it to their local environment. 

For all the collapsible code sections, there is always a file name in the box's title. These are the codes that users need to write down in their local repository as part of the tutorial.
![image](https://github.com/knative/docs/assets/36619969/a8cfaea6-16e0-4faf-bb4b-adcf6dc5a785)

The collapsible feature is used to distinguish between files that are part of the tutorial and those that are only there to illustrate concepts. @creydr

Reply:
I mean it compared to something without this box arround. E.g. the code samples in https://deploy-preview-6007--knative.netlify.app/docs/bookstore/page-1/send-review-comment-to-broker/#step-1-understand-book-review-service or how we did it usually in the documentation: https://knative.dev/docs/eventing/brokers/create-broker/#broker-class-options

Reply:
Yes I see that.

![image](https://github.com/knative/docs/assets/36619969/bed3a4da-ee96-456c-a52a-8ca2e1224446)

I designed it so that there are 2 ways to showcase the code in Sample App. The new collapsible box format highlights important code snippets. This ensures that readers instantly recognize the significance of the code. Also, the file name is displayed at the top of the box, guiding users on where to place the copied code. Once users complete the task, they can collapse the box, marking it as completed—much like checking off an item on a to-do list. Hope that clear the confusion! @creydr

Reply:
makes sense then. :+1: